src/widgetastic_patternfly5/components/switch.py

```diff
@@ -26,32 +26,36 @@ def click(self):
         """Click on a Switch."""
         if not self.is_enabled:
             raise SwitchDisabled(f"{repr(self)} is disabled")
-        else:
-            self.browser.click(self.CHECKBOX_LOCATOR)
-            return True
+
+        self.browser.click(self.CHECKBOX_LOCATOR)
+        return True
 
     @property
     def selected(self):
         """Returns a boolean detailing if the Switch is on (True) of off (False)."""
-        return self.browser.get_attribute("checked", self.CHECKBOX_LOCATOR) is not None
+        return self.browser.is_checked(self.CHECKBOX_LOCATOR)
 
     def fill(self, value):
         """Fills a Switch with the supplied value."""
         if not self.is_enabled:
             raise SwitchDisabled(f"{repr(self)} is disabled")
         if bool(value) == self.selected:
             return False
+
+        el = self.browser.element(self.CHECKBOX_LOCATOR)
+        if value == True:
```

Comment:
Why an explicit check against `True` and not `if value` allowing truthy setting of the checkbox.